2025-08-29 14:59:26.659 | INFO     | Watching for file changes with StatReloader
2025-08-29 14:59:27.005 | INFO     | 
2025-08-29 14:59:27.005 | INFO     | --- Start APP ---
2025-08-29 14:59:27.005 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 14:59:27.009 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:04:21.087 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:04:21.208 | INFO     | 
2025-08-29 15:04:21.208 | INFO     | --- Start APP ---
2025-08-29 15:04:21.208 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:04:21.219 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:04:29.517 | SUCCESS  | Ping des MongoDB-Servers war erfolgreich
2025-08-29 15:04:29.522 | INFO     | Konfigurationsdatei für die Datenbankverbindung aktualisiert
2025-08-29 15:04:29.522 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 15:06:30.445 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:06:30.555 | INFO     | 
2025-08-29 15:06:30.555 | INFO     | --- Start APP ---
2025-08-29 15:06:30.555 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:06:30.563 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:06:58.961 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:06:59.067 | INFO     | 
2025-08-29 15:06:59.067 | INFO     | --- Start APP ---
2025-08-29 15:06:59.067 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:06:59.076 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:07:39.176 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:07:39.285 | INFO     | 
2025-08-29 15:07:39.285 | INFO     | --- Start APP ---
2025-08-29 15:07:39.285 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:07:39.294 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:07:50.509 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:07:50.615 | INFO     | 
2025-08-29 15:07:50.615 | INFO     | --- Start APP ---
2025-08-29 15:07:50.615 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:07:50.623 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:09:20.008 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:09:20.112 | INFO     | 
2025-08-29 15:09:20.112 | INFO     | --- Start APP ---
2025-08-29 15:09:20.112 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:09:20.122 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:09:26.549 | SUCCESS  | Ping des MongoDB-Servers war erfolgreich
2025-08-29 15:09:26.555 | INFO     | Konfigurationsdatei für die Datenbankverbindung aktualisiert
2025-08-29 15:09:26.556 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 15:12:11.890 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:12:11.994 | INFO     | 
2025-08-29 15:12:11.994 | INFO     | --- Start APP ---
2025-08-29 15:12:12.036 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 15:15:16.077 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:15:16.185 | INFO     | 
2025-08-29 15:15:16.185 | INFO     | --- Start APP ---
2025-08-29 15:15:16.252 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 15:15:30.017 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:15:30.131 | INFO     | 
2025-08-29 15:15:30.131 | INFO     | --- Start APP ---
2025-08-29 15:15:30.131 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:15:30.138 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:16:38.117 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:16:38.228 | INFO     | 
2025-08-29 15:16:38.228 | INFO     | --- Start APP ---
2025-08-29 15:16:38.228 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:16:38.235 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:18:50.530 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:18:50.634 | INFO     | 
2025-08-29 15:18:50.634 | INFO     | --- Start APP ---
2025-08-29 15:18:50.634 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:18:50.646 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:34:01.166 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:34:01.302 | INFO     | 
2025-08-29 15:34:01.303 | INFO     | --- Start APP ---
2025-08-29 15:34:01.303 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:34:01.309 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:34:01.363 | WARNING  | Not Found: /mongodb/mongo_connection/toasts/toasts.css
2025-08-29 15:34:01.363 | WARNING  | Not Found: /mongodb/mongo_connection/toasts/toasts.js'></script><div id=
2025-08-29 15:43:07.338 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:43:07.461 | INFO     | 
2025-08-29 15:43:07.462 | INFO     | --- Start APP ---
2025-08-29 15:43:07.462 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:43:07.466 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:43:29.070 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1ae7b2a1c8a6d69638201, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 15:43:29.091 | ERROR    | Internal Server Error: /mongodb/mongo_connection/
Traceback (most recent call last):

  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1032, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x101cc1940>
    └ <Thread(Thread-3 (process_request_thread), started daemon 6192033792)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x101cc1620>
    └ <Thread(Thread-3 (process_request_thread), started daemon 6192033792)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-3 (process_request_thread), started daemon 6192033792)>
    │    │        │    └ (<socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50279)>, ('127.0.0.1', 50279))
    │    │        └ <Thread(Thread-3 (process_request_thread), started daemon 6192033792)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <django.core.servers.basehttp.WSGIServer object at 0x103f06c00>>
    └ <Thread(Thread-3 (process_request_thread), started daemon 6192033792)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 50279)
    │    │              └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50279)>
    │    └ <function BaseServer.finish_request at 0x1031e1260>
    └ <django.core.servers.basehttp.WSGIServer object at 0x103f06c00>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <django.core.servers.basehttp.WSGIServer object at 0x103f06c00>
    │    │                   │        └ ('127.0.0.1', 50279)
    │    │                   └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50279)>
    │    └ <class 'django.core.servers.basehttp.WSGIRequestHandler'>
    └ <django.core.servers.basehttp.WSGIServer object at 0x103f06c00>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x10380f9c0>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1055a2e10>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 228, in handle
    self.handle_one_request()
    │    └ <function WSGIRequestHandler.handle_one_request at 0x10380fa60>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1055a2e10>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 253, in handle_one_request
    handler.run(self.server.get_app())
    │       │   │    │      └ <function WSGIServer.get_app at 0x10380eac0>
    │       │   │    └ <django.core.servers.basehttp.WSGIServer object at 0x103f06c00>
    │       │   └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1055a2e10>
    │       └ <function BaseHandler.run at 0x103806ac0>
    └ <django.core.servers.basehttp.ServerHandler object at 0x10552c6e0>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/wsgiref/handlers.py", line 137, in run
    self.result = application(self.environ, self.start_response)
    │    │        │           │    │        │    └ <function BaseHandler.start_response at 0x103806e80>
    │    │        │           │    │        └ <django.core.servers.basehttp.ServerHandler object at 0x10552c6e0>
    │    │        │           │    └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
    │    │        │           └ <django.core.servers.basehttp.ServerHandler object at 0x10552c6e0>
    │    │        └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x1051991c0>
    │    └ None
    └ <django.core.servers.basehttp.ServerHandler object at 0x10552c6e0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/contrib/staticfiles/handlers.py", line 80, in __call__
    return self.application(environ, start_response)
           │    │           │        └ <bound method BaseHandler.start_response of <django.core.servers.basehttp.ServerHandler object at 0x10552c6e0>>
           │    │           └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
           │    └ <django.core.handlers.wsgi.WSGIHandler object at 0x102a4a090>
           └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x1051991c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
    response = self.get_response(request)
               │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function BaseHandler.get_response at 0x1037cc5e0>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x102a4a090>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
               │    │                 └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function convert_exception_to_response.<locals>.inner at 0x10523b560>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x102a4a090>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x10523b380>
               │           └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x10523b2e0>
               │           └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x10523b240>
               │           └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x10523b1a0>
               │           └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x10523ade0>
               │           └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x10523ad40>
               │           └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function BaseHandler._get_response at 0x1038245e0>
               │           └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
               └ None
> File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <bound method BaseHandler._get_response of <django.core.handlers.wsgi.WSGIHandler object at 0x102a4a090>>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               │                │         │                └ {}
               │                │         └ ()
               │                └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <function mongo_connection at 0x10518a8e0>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 53, in mongo_connection
    return render_toast_response(request)
           │                     └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
           └ <function render_toast_response at 0x10518a840>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 17, in render_toast_response
    html = render_to_string("partials/toasts.html", {"messages": storage})
           │                                                     └ <FallbackStorage: request=<WSGIRequest: POST '/mongodb/mongo_connection/'>>
           └ <function render_to_string at 0x102d04f40>

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               │            │                    └ None
               │            └ 'partials/toasts.html'
               └ <function get_template at 0x102b834c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
          │                    │                    └ [TemplateDoesNotExist('partials/toasts.html')]
          │                    └ 'partials/toasts.html'
          └ <class 'django.template.exceptions.TemplateDoesNotExist'>

django.template.exceptions.TemplateDoesNotExist: partials/toasts.html
2025-08-29 15:44:13.018 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:44:13.129 | INFO     | 
2025-08-29 15:44:13.129 | INFO     | --- Start APP ---
2025-08-29 15:44:13.129 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:44:13.137 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:44:34.039 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1aebc1a7b589304f1a2a2, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 15:44:34.058 | ERROR    | Internal Server Error: /mongodb/mongo_connection/
Traceback (most recent call last):

  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1032, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x1018e1940>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6191378432)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x1018e1620>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6191378432)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6191378432)>
    │    │        │    └ (<socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50315)>, ('127.0.0.1', 50315))
    │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6191378432)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6191378432)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 50315)
    │    │              └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50315)>
    │    └ <function BaseServer.finish_request at 0x10322d260>
    └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
    │    │                   │        └ ('127.0.0.1', 50315)
    │    │                   └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50315)>
    │    └ <class 'django.core.servers.basehttp.WSGIRequestHandler'>
    └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x10368b9c0>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x103ec77d0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 230, in handle
    self.handle_one_request()
    │    └ <function WSGIRequestHandler.handle_one_request at 0x10368ba60>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x103ec77d0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 253, in handle_one_request
    handler.run(self.server.get_app())
    │       │   │    │      └ <function WSGIServer.get_app at 0x10368aac0>
    │       │   │    └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
    │       │   └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x103ec77d0>
    │       └ <function BaseHandler.run at 0x103682ac0>
    └ <django.core.servers.basehttp.ServerHandler object at 0x103dbcad0>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/wsgiref/handlers.py", line 137, in run
    self.result = application(self.environ, self.start_response)
    │    │        │           │    │        │    └ <function BaseHandler.start_response at 0x103682e80>
    │    │        │           │    │        └ <django.core.servers.basehttp.ServerHandler object at 0x103dbcad0>
    │    │        │           │    └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
    │    │        │           └ <django.core.servers.basehttp.ServerHandler object at 0x103dbcad0>
    │    │        └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x1039f7fe0>
    │    └ None
    └ <django.core.servers.basehttp.ServerHandler object at 0x103dbcad0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/contrib/staticfiles/handlers.py", line 80, in __call__
    return self.application(environ, start_response)
           │    │           │        └ <bound method BaseHandler.start_response of <django.core.servers.basehttp.ServerHandler object at 0x103dbcad0>>
           │    │           └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
           │    └ <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>
           └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x1039f7fe0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
    response = self.get_response(request)
               │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function BaseHandler.get_response at 0x1036485e0>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
               │    │                 └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37f60>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37d80>
               │           └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37ce0>
               │           └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37c40>
               │           └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37ba0>
               │           └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37920>
               │           └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37b00>
               │           └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function BaseHandler._get_response at 0x1036a05e0>
               │           └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
               └ None
> File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <bound method BaseHandler._get_response of <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               │                │         │                └ {}
               │                │         └ ()
               │                └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <function mongo_connection at 0x103dae8e0>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 53, in mongo_connection
    return render_toast_response(request)
           │                     └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
           └ <function render_toast_response at 0x103dae840>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 17, in render_toast_response
    html = render_to_string("partials/toasts.html", {"messages": storage})
           │                                                     └ <FallbackStorage: request=<WSGIRequest: POST '/mongodb/mongo_connection/'>>
           └ <function render_to_string at 0x102d50f40>

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               │            │                    └ None
               │            └ 'partials/toasts.html'
               └ <function get_template at 0x102bcf4c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
          │                    │                    └ [TemplateDoesNotExist('partials/toasts.html')]
          │                    └ 'partials/toasts.html'
          └ <class 'django.template.exceptions.TemplateDoesNotExist'>

django.template.exceptions.TemplateDoesNotExist: partials/toasts.html
2025-08-29 15:44:42.413 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1aec51a7b589304f1a2a3, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 15:44:42.431 | ERROR    | Internal Server Error: /mongodb/mongo_connection/
Traceback (most recent call last):

  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1032, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x1018e1940>
    └ <Thread(Thread-2 (process_request_thread), started daemon 6208204800)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x1018e1620>
    └ <Thread(Thread-2 (process_request_thread), started daemon 6208204800)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-2 (process_request_thread), started daemon 6208204800)>
    │    │        │    └ (<socket.socket fd=9, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50316)>, ('127.0.0.1', 50316))
    │    │        └ <Thread(Thread-2 (process_request_thread), started daemon 6208204800)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>>
    └ <Thread(Thread-2 (process_request_thread), started daemon 6208204800)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 50316)
    │    │              └ <socket.socket fd=9, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50316)>
    │    └ <function BaseServer.finish_request at 0x10322d260>
    └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
    │    │                   │        └ ('127.0.0.1', 50316)
    │    │                   └ <socket.socket fd=9, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50316)>
    │    └ <class 'django.core.servers.basehttp.WSGIRequestHandler'>
    └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x10368b9c0>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x10415b740>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 230, in handle
    self.handle_one_request()
    │    └ <function WSGIRequestHandler.handle_one_request at 0x10368ba60>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x10415b740>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 253, in handle_one_request
    handler.run(self.server.get_app())
    │       │   │    │      └ <function WSGIServer.get_app at 0x10368aac0>
    │       │   │    └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
    │       │   └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x10415b740>
    │       └ <function BaseHandler.run at 0x103682ac0>
    └ <django.core.servers.basehttp.ServerHandler object at 0x103dbcb30>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/wsgiref/handlers.py", line 137, in run
    self.result = application(self.environ, self.start_response)
    │    │        │           │    │        │    └ <function BaseHandler.start_response at 0x103682e80>
    │    │        │           │    │        └ <django.core.servers.basehttp.ServerHandler object at 0x103dbcb30>
    │    │        │           │    └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
    │    │        │           └ <django.core.servers.basehttp.ServerHandler object at 0x103dbcb30>
    │    │        └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x1039f7fe0>
    │    └ None
    └ <django.core.servers.basehttp.ServerHandler object at 0x103dbcb30>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/contrib/staticfiles/handlers.py", line 80, in __call__
    return self.application(environ, start_response)
           │    │           │        └ <bound method BaseHandler.start_response of <django.core.servers.basehttp.ServerHandler object at 0x103dbcb30>>
           │    │           └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
           │    └ <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>
           └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x1039f7fe0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
    response = self.get_response(request)
               │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function BaseHandler.get_response at 0x1036485e0>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
               │    │                 └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37f60>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37d80>
               │           └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37ce0>
               │           └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37c40>
               │           └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37ba0>
               │           └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37920>
               │           └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37b00>
               │           └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function BaseHandler._get_response at 0x1036a05e0>
               │           └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
               └ None
> File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <bound method BaseHandler._get_response of <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               │                │         │                └ {}
               │                │         └ ()
               │                └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <function mongo_connection at 0x103dae8e0>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 53, in mongo_connection
    return render_toast_response(request)
           │                     └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
           └ <function render_toast_response at 0x103dae840>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 17, in render_toast_response
    html = render_to_string("partials/toasts.html", {"messages": storage})
           │                                                     └ <FallbackStorage: request=<WSGIRequest: POST '/mongodb/mongo_connection/'>>
           └ <function render_to_string at 0x102d50f40>

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               │            │                    └ None
               │            └ 'partials/toasts.html'
               └ <function get_template at 0x102bcf4c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
          │                    │                    └ [TemplateDoesNotExist('partials/toasts.html')]
          │                    └ 'partials/toasts.html'
          └ <class 'django.template.exceptions.TemplateDoesNotExist'>

django.template.exceptions.TemplateDoesNotExist: partials/toasts.html
2025-08-29 15:44:45.874 | SUCCESS  | Ping des MongoDB-Servers war erfolgreich
2025-08-29 15:44:45.880 | INFO     | Konfigurationsdatei für die Datenbankverbindung aktualisiert
2025-08-29 15:44:45.880 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 15:44:45.892 | ERROR    | Internal Server Error: /mongodb/mongo_connection/
Traceback (most recent call last):

  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1032, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x1018e1940>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6191378432)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x1018e1620>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6191378432)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6191378432)>
    │    │        │    └ (<socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50315)>, ('127.0.0.1', 50315))
    │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6191378432)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6191378432)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 50315)
    │    │              └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50315)>
    │    └ <function BaseServer.finish_request at 0x10322d260>
    └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
    │    │                   │        └ ('127.0.0.1', 50315)
    │    │                   └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50315)>
    │    └ <class 'django.core.servers.basehttp.WSGIRequestHandler'>
    └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x10368b9c0>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x103ec77d0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 230, in handle
    self.handle_one_request()
    │    └ <function WSGIRequestHandler.handle_one_request at 0x10368ba60>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x103ec77d0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 253, in handle_one_request
    handler.run(self.server.get_app())
    │       │   │    │      └ <function WSGIServer.get_app at 0x10368aac0>
    │       │   │    └ <django.core.servers.basehttp.WSGIServer object at 0x103e26e40>
    │       │   └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x103ec77d0>
    │       └ <function BaseHandler.run at 0x103682ac0>
    └ <django.core.servers.basehttp.ServerHandler object at 0x103f0baa0>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/wsgiref/handlers.py", line 137, in run
    self.result = application(self.environ, self.start_response)
    │    │        │           │    │        │    └ <function BaseHandler.start_response at 0x103682e80>
    │    │        │           │    │        └ <django.core.servers.basehttp.ServerHandler object at 0x103f0baa0>
    │    │        │           │    └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
    │    │        │           └ <django.core.servers.basehttp.ServerHandler object at 0x103f0baa0>
    │    │        └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x1039f7fe0>
    │    └ None
    └ <django.core.servers.basehttp.ServerHandler object at 0x103f0baa0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/contrib/staticfiles/handlers.py", line 80, in __call__
    return self.application(environ, start_response)
           │    │           │        └ <bound method BaseHandler.start_response of <django.core.servers.basehttp.ServerHandler object at 0x103f0baa0>>
           │    │           └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
           │    └ <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>
           └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x1039f7fe0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
    response = self.get_response(request)
               │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function BaseHandler.get_response at 0x1036485e0>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
               │    │                 └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37f60>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37d80>
               │           └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37ce0>
               │           └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37c40>
               │           └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37ba0>
               │           └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37920>
               │           └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103e37b00>
               │           └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function BaseHandler._get_response at 0x1036a05e0>
               │           └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
               └ None
> File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <bound method BaseHandler._get_response of <django.core.handlers.wsgi.WSGIHandler object at 0x1012bd4c0>>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               │                │         │                └ {}
               │                │         └ ()
               │                └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <function mongo_connection at 0x103dae8e0>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 48, in mongo_connection
    return render_toast_response(request)
           │                     └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
           └ <function render_toast_response at 0x103dae840>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 17, in render_toast_response
    html = render_to_string("partials/toasts.html", {"messages": storage})
           │                                                     └ <FallbackStorage: request=<WSGIRequest: POST '/mongodb/mongo_connection/'>>
           └ <function render_to_string at 0x102d50f40>

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               │            │                    └ None
               │            └ 'partials/toasts.html'
               └ <function get_template at 0x102bcf4c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
          │                    │                    └ [TemplateDoesNotExist('partials/toasts.html')]
          │                    └ 'partials/toasts.html'
          └ <class 'django.template.exceptions.TemplateDoesNotExist'>

django.template.exceptions.TemplateDoesNotExist: partials/toasts.html
2025-08-29 15:46:02.059 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:46:02.176 | INFO     | 
2025-08-29 15:46:02.176 | INFO     | --- Start APP ---
2025-08-29 15:46:02.205 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 15:51:57.687 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:51:57.809 | INFO     | 
2025-08-29 15:51:57.809 | INFO     | --- Start APP ---
2025-08-29 15:51:57.838 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 15:52:07.452 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:52:07.553 | INFO     | 
2025-08-29 15:52:07.553 | INFO     | --- Start APP ---
2025-08-29 15:52:07.553 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:52:07.565 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:52:20.293 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b08f199ced3ac5853823, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 15:52:43.486 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b0a6199ced3ac5853824, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 15:57:49.345 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:57:49.453 | INFO     | 
2025-08-29 15:57:49.453 | INFO     | --- Start APP ---
2025-08-29 15:57:49.453 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:57:49.459 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:58:01.293 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b1e40a6bb8ef4b697244, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 15:58:19.492 | INFO     | Watching for file changes with StatReloader
2025-08-29 15:58:19.592 | INFO     | 
2025-08-29 15:58:19.592 | INFO     | --- Start APP ---
2025-08-29 15:58:19.592 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 15:58:19.601 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:00:37.736 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:00:37.840 | INFO     | 
2025-08-29 16:00:37.840 | INFO     | --- Start APP ---
2025-08-29 16:00:37.840 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:00:37.850 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:00:46.949 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27018: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b289f7f40dbb49d75a05, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27018: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 16:03:00.144 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:03:00.255 | INFO     | 
2025-08-29 16:03:00.255 | INFO     | --- Start APP ---
2025-08-29 16:03:00.255 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:03:12.453 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b31bdb00a3f0ad9e4139, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 16:05:57.500 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:05:57.609 | INFO     | 
2025-08-29 16:05:57.609 | INFO     | --- Start APP ---
2025-08-29 16:05:57.609 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:06:09.826 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b3cce5188edde26d5102, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 16:07:28.263 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:07:28.373 | INFO     | 
2025-08-29 16:07:28.373 | INFO     | --- Start APP ---
2025-08-29 16:07:28.373 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:07:38.317 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.localй:27017: [Errno 8] nodename nor servname provided, or not known (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b4258abb1cf00d2ba89d, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.localй', 27017) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.localй:27017: [Errno 8] nodename nor servname provided, or not known (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 16:10:10.494 | INFO     | /Users/evgenij/Projects/WWS1/mongodb/views.py changed, reloading.
2025-08-29 16:10:10.665 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:11:12.999 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:11:13.115 | INFO     | 
2025-08-29 16:11:13.115 | INFO     | --- Start APP ---
2025-08-29 16:11:13.115 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:11:13.122 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:11:23.084 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27018: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b50636ab82184f606167, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27018: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 16:14:47.849 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:14:47.963 | INFO     | 
2025-08-29 16:14:47.963 | INFO     | --- Start APP ---
2025-08-29 16:14:47.963 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:14:47.973 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:14:57.166 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b5dcc6d73d69c60f02ae, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 16:16:53.899 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:16:54.006 | INFO     | 
2025-08-29 16:16:54.006 | INFO     | --- Start APP ---
2025-08-29 16:16:54.006 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:16:54.014 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:17:05.102 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b65c83315d7578d1f043, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 16:19:10.725 | INFO     | /Users/evgenij/Projects/WWS1/mongodb/views.py changed, reloading.
2025-08-29 16:19:10.981 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:21:31.065 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:21:31.194 | INFO     | 
2025-08-29 16:21:31.194 | INFO     | --- Start APP ---
2025-08-29 16:21:31.194 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:21:31.201 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:21:40.413 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b76f60149993d6451a66, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 16:22:52.226 | INFO     | /Users/evgenij/Projects/WWS1/mongodb/views.py changed, reloading.
2025-08-29 16:22:52.469 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:24:06.390 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:24:06.517 | INFO     | 
2025-08-29 16:24:06.517 | INFO     | --- Start APP ---
2025-08-29 16:24:06.517 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:24:31.041 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:24:31.140 | INFO     | 
2025-08-29 16:24:31.140 | INFO     | --- Start APP ---
2025-08-29 16:24:31.140 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:24:49.885 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:24:49.997 | INFO     | 
2025-08-29 16:24:49.997 | INFO     | --- Start APP ---
2025-08-29 16:24:49.997 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:25:10.605 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:25:10.718 | INFO     | 
2025-08-29 16:25:10.718 | INFO     | --- Start APP ---
2025-08-29 16:25:10.718 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:25:27.369 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:25:27.476 | INFO     | 
2025-08-29 16:25:27.476 | INFO     | --- Start APP ---
2025-08-29 16:25:27.476 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:25:27.484 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:28:45.076 | INFO     | /Users/evgenij/Projects/WWS1/mongodb/views.py changed, reloading.
2025-08-29 16:28:45.265 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:30:54.732 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:30:54.848 | INFO     | 
2025-08-29 16:30:54.848 | INFO     | --- Start APP ---
2025-08-29 16:30:54.848 | ERROR    | Ошибка расшифровки файла конфигурации: 
2025-08-29 16:30:54.848 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:30:54.854 | ERROR    | Ошибка расшифровки файла конфигурации: 
2025-08-29 16:30:54.854 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:30:54.872 | ERROR    | Internal Server Error: /mongodb/mongo_connection/
Traceback (most recent call last):

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/backends/django.py", line 107, in render
    return self.template.render(context)
           │    │        │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;p...
           │    │        └ <function Template.render at 0x102428d60>
           │    └ <Template template_string="{% extends 'base.htm...">
           └ <django.template.backends.django.Template object at 0x103259760>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           │    │       └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;p...
           │    └ <function Template._render at 0x102428cc0>
           └ <Template template_string="{% extends 'base.htm...">
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           │    │        │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;p...
           │    │        └ <function NodeList.render at 0x10242a8e0>
           │    └ [<ExtendsNode: extends 'base.html'>]
           └ <Template template_string="{% extends 'base.htm...">
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
           │                                         │                    └ [<ExtendsNode: extends 'base.html'>]
           │                                         └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;p...
           └ <class 'django.utils.safestring.SafeString'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 977, in render_annotated
    return self.render(context)
           │    │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;p...
           │    └ <function ExtendsNode.render at 0x102a44540>
           └ <ExtendsNode: extends 'base.html'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           │               │       └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;p...
           │               └ <function Template._render at 0x102428cc0>
           └ <Template template_string="{% load static %}<!...">
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           │    │        │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;p...
           │    │        └ <function NodeList.render at 0x10242a8e0>
           │    └ [<django.template.defaulttags.LoadNode object at 0x104210fb0>, <TextNode: '\n<!DOCTYPE html>\n<html la'>, <Block Node: title....
           └ <Template template_string="{% load static %}<!...">
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
           │                                         │                    └ [<django.template.defaulttags.LoadNode object at 0x104210fb0>, <TextNode: '\n<!DOCTYPE html>\n<html la'>, <Block Node: title....
           │                                         └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;p...
           └ <class 'django.utils.safestring.SafeString'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 977, in render_annotated
    return self.render(context)
           │    │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;p...
           │    └ <function IncludeNode.render at 0x102a44720>
           └ <IncludeNode: template=<FilterExpression "'partials/toasts.html'">>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader_tags.py", line 199, in render
    template = context.template.engine.select_template(template_name)
               │       │                               └ ('partials/toasts.html',)
               │       └ None
               └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;p...
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/engine.py", line 214, in select_template
    raise TemplateDoesNotExist(", ".join(not_found))
          │                              └ ['partials/toasts.html']
          └ <class 'django.template.exceptions.TemplateDoesNotExist'>

django.template.exceptions.TemplateDoesNotExist: partials/toasts.html


The above exception was the direct cause of the following exception:


Traceback (most recent call last):

  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1032, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x101429940>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6201012224)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x101429620>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6201012224)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6201012224)>
    │    │        │    └ (<socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50610)>, ('127.0.0.1', 50610))
    │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6201012224)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <django.core.servers.basehttp.WSGIServer object at 0x1030fed20>>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6201012224)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 50610)
    │    │              └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50610)>
    │    └ <function BaseServer.finish_request at 0x10290d260>
    └ <django.core.servers.basehttp.WSGIServer object at 0x1030fed20>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <django.core.servers.basehttp.WSGIServer object at 0x1030fed20>
    │    │                   │        └ ('127.0.0.1', 50610)
    │    │                   └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50610)>
    │    └ <class 'django.core.servers.basehttp.WSGIRequestHandler'>
    └ <django.core.servers.basehttp.WSGIServer object at 0x1030fed20>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x102d6b9c0>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1037ab6b0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 230, in handle
    self.handle_one_request()
    │    └ <function WSGIRequestHandler.handle_one_request at 0x102d6ba60>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1037ab6b0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 253, in handle_one_request
    handler.run(self.server.get_app())
    │       │   │    │      └ <function WSGIServer.get_app at 0x102d6aac0>
    │       │   │    └ <django.core.servers.basehttp.WSGIServer object at 0x1030fed20>
    │       │   └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1037ab6b0>
    │       └ <function BaseHandler.run at 0x102d62ac0>
    └ <django.core.servers.basehttp.ServerHandler object at 0x103600e00>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/wsgiref/handlers.py", line 137, in run
    self.result = application(self.environ, self.start_response)
    │    │        │           │    │        │    └ <function BaseHandler.start_response at 0x102d62e80>
    │    │        │           │    │        └ <django.core.servers.basehttp.ServerHandler object at 0x103600e00>
    │    │        │           │    └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
    │    │        │           └ <django.core.servers.basehttp.ServerHandler object at 0x103600e00>
    │    │        └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x102cca330>
    │    └ None
    └ <django.core.servers.basehttp.ServerHandler object at 0x103600e00>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/contrib/staticfiles/handlers.py", line 80, in __call__
    return self.application(environ, start_response)
           │    │           │        └ <bound method BaseHandler.start_response of <django.core.servers.basehttp.ServerHandler object at 0x103600e00>>
           │    │           └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
           │    └ <django.core.handlers.wsgi.WSGIHandler object at 0x102682c60>
           └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x102cca330>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
    response = self.get_response(request)
               │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │    └ <function BaseHandler.get_response at 0x102d285e0>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x102682c60>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
               │    │                 └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │    └ <function convert_exception_to_response.<locals>.inner at 0x103713f60>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x102682c60>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103713ec0>
               │           └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103713d80>
               │           └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103713ce0>
               │           └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103713c40>
               │           └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103713ba0>
               │           └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x103713b00>
               │           └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function BaseHandler._get_response at 0x102d805e0>
               │           └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
               └ None
> File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <bound method BaseHandler._get_response of <django.core.handlers.wsgi.WSGIHandler object at 0x102682c60>>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               │                │         │                └ {}
               │                │         └ ()
               │                └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <function mongo_connection at 0x10368a8e0>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 80, in mongo_connection
    return render(request, 'mongodb/connection_form.html', {
           │      └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
           └ <function render at 0x102b01bc0>

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              │      │                │              │        │              └ None
              │      │                │              │        └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
              │      │                │              └ {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;port)>, 'text': {'title': 'Serverkonfiguration', 'header...
              │      │                └ 'mongodb/connection_form.html'
              │      └ <function render_to_string at 0x102430f40>
              └ <module 'django.template.loader' from '/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader...
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           │        │      │        └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
           │        │      └ {'form': <MongoConnectionForm bound=False, valid=False, fields=(host;port)>, 'text': {'title': 'Serverkonfiguration', 'header...
           │        └ <function Template.render at 0x102432200>
           └ <django.template.backends.django.Template object at 0x103259760>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/backends/django.py", line 109, in render
    reraise(exc, self.backend)
    │            │    └ <django.template.backends.django.DjangoTemplates object at 0x102d75010>
    │            └ <django.template.backends.django.Template object at 0x103259760>
    └ <function reraise at 0x1024322a0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/backends/django.py", line 130, in reraise
    raise new from exc
          │        └ TemplateDoesNotExist('partials/toasts.html')
          └ TemplateDoesNotExist('partials/toasts.html')

django.template.exceptions.TemplateDoesNotExist: partials/toasts.html
2025-08-29 16:31:50.817 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:31:50.930 | INFO     | 
2025-08-29 16:31:50.930 | INFO     | --- Start APP ---
2025-08-29 16:31:50.930 | ERROR    | Ошибка расшифровки файла конфигурации: 
2025-08-29 16:31:50.930 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:31:50.937 | ERROR    | Ошибка расшифровки файла конфигурации: 
2025-08-29 16:31:50.937 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:31:52.620 | SUCCESS  | Ping des MongoDB-Servers war erfolgreich
2025-08-29 16:31:52.620 | ERROR    | Ошибка расшифровки файла конфигурации: 
2025-08-29 16:31:52.622 | INFO     | Konfigurationsdatei für die Datenbankverbindung aktualisiert
2025-08-29 16:31:52.622 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 16:31:52.628 | ERROR    | Internal Server Error: /mongodb/mongo_connection/
Traceback (most recent call last):

  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1032, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x102da5940>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6169653248)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x102da5620>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6169653248)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6169653248)>
    │    │        │    └ (<socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50617)>, ('127.0.0.1', 50617))
    │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6169653248)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <django.core.servers.basehttp.WSGIServer object at 0x1051472c0>>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6169653248)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 50617)
    │    │              └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50617)>
    │    └ <function BaseServer.finish_request at 0x104715260>
    └ <django.core.servers.basehttp.WSGIServer object at 0x1051472c0>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <django.core.servers.basehttp.WSGIServer object at 0x1051472c0>
    │    │                   │        └ ('127.0.0.1', 50617)
    │    │                   └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50617)>
    │    └ <class 'django.core.servers.basehttp.WSGIRequestHandler'>
    └ <django.core.servers.basehttp.WSGIServer object at 0x1051472c0>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x104b739c0>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1055a3b00>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 230, in handle
    self.handle_one_request()
    │    └ <function WSGIRequestHandler.handle_one_request at 0x104b73a60>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1055a3b00>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 253, in handle_one_request
    handler.run(self.server.get_app())
    │       │   │    │      └ <function WSGIServer.get_app at 0x104b72ac0>
    │       │   │    └ <django.core.servers.basehttp.WSGIServer object at 0x1051472c0>
    │       │   └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1055a3b00>
    │       └ <function BaseHandler.run at 0x104b6aac0>
    └ <django.core.servers.basehttp.ServerHandler object at 0x10600eea0>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/wsgiref/handlers.py", line 137, in run
    self.result = application(self.environ, self.start_response)
    │    │        │           │    │        │    └ <function BaseHandler.start_response at 0x104b6ae80>
    │    │        │           │    │        └ <django.core.servers.basehttp.ServerHandler object at 0x10600eea0>
    │    │        │           │    └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
    │    │        │           └ <django.core.servers.basehttp.ServerHandler object at 0x10600eea0>
    │    │        └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x10545fb30>
    │    └ None
    └ <django.core.servers.basehttp.ServerHandler object at 0x10600eea0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/contrib/staticfiles/handlers.py", line 80, in __call__
    return self.application(environ, start_response)
           │    │           │        └ <bound method BaseHandler.start_response of <django.core.servers.basehttp.ServerHandler object at 0x10600eea0>>
           │    │           └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
           │    └ <django.core.handlers.wsgi.WSGIHandler object at 0x1046c2720>
           └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x10545fb30>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
    response = self.get_response(request)
               │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function BaseHandler.get_response at 0x104b305e0>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1046c2720>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
               │    │                 └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function convert_exception_to_response.<locals>.inner at 0x105513f60>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1046c2720>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513ec0>
               │           └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513d80>
               │           └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513ce0>
               │           └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513c40>
               │           └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513ba0>
               │           └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513b00>
               │           └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function BaseHandler._get_response at 0x104b885e0>
               │           └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
               └ None
> File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <bound method BaseHandler._get_response of <django.core.handlers.wsgi.WSGIHandler object at 0x1046c2720>>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               │                │         │                └ {}
               │                │         └ ()
               │                └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <function mongo_connection at 0x10548a8e0>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 48, in mongo_connection
    return render_toast_response(request)
           │                     └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
           └ <function render_toast_response at 0x10548a840>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 17, in render_toast_response
    html = render_to_string("partials/toasts.html", {"messages": storage})
           │                                                     └ <FallbackStorage: request=<WSGIRequest: POST '/mongodb/mongo_connection/'>>
           └ <function render_to_string at 0x104238f40>

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               │            │                    └ None
               │            └ 'partials/toasts.html'
               └ <function get_template at 0x1040b74c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
          │                    │                    └ [TemplateDoesNotExist('partials/toasts.html')]
          │                    └ 'partials/toasts.html'
          └ <class 'django.template.exceptions.TemplateDoesNotExist'>

django.template.exceptions.TemplateDoesNotExist: partials/toasts.html
2025-08-29 16:32:02.492 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1b9dd24f82d8ac4a100c1, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 16:32:02.507 | ERROR    | Internal Server Error: /mongodb/mongo_connection/
Traceback (most recent call last):

  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1032, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x102da5940>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6169653248)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x102da5620>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6169653248)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6169653248)>
    │    │        │    └ (<socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50617)>, ('127.0.0.1', 50617))
    │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6169653248)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <django.core.servers.basehttp.WSGIServer object at 0x1051472c0>>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6169653248)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 50617)
    │    │              └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50617)>
    │    └ <function BaseServer.finish_request at 0x104715260>
    └ <django.core.servers.basehttp.WSGIServer object at 0x1051472c0>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <django.core.servers.basehttp.WSGIServer object at 0x1051472c0>
    │    │                   │        └ ('127.0.0.1', 50617)
    │    │                   └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50617)>
    │    └ <class 'django.core.servers.basehttp.WSGIRequestHandler'>
    └ <django.core.servers.basehttp.WSGIServer object at 0x1051472c0>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x104b739c0>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1055a3b00>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 230, in handle
    self.handle_one_request()
    │    └ <function WSGIRequestHandler.handle_one_request at 0x104b73a60>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1055a3b00>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 253, in handle_one_request
    handler.run(self.server.get_app())
    │       │   │    │      └ <function WSGIServer.get_app at 0x104b72ac0>
    │       │   │    └ <django.core.servers.basehttp.WSGIServer object at 0x1051472c0>
    │       │   └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x1055a3b00>
    │       └ <function BaseHandler.run at 0x104b6aac0>
    └ <django.core.servers.basehttp.ServerHandler object at 0x1051ddfa0>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/wsgiref/handlers.py", line 137, in run
    self.result = application(self.environ, self.start_response)
    │    │        │           │    │        │    └ <function BaseHandler.start_response at 0x104b6ae80>
    │    │        │           │    │        └ <django.core.servers.basehttp.ServerHandler object at 0x1051ddfa0>
    │    │        │           │    └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
    │    │        │           └ <django.core.servers.basehttp.ServerHandler object at 0x1051ddfa0>
    │    │        └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x10545fb30>
    │    └ None
    └ <django.core.servers.basehttp.ServerHandler object at 0x1051ddfa0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/contrib/staticfiles/handlers.py", line 80, in __call__
    return self.application(environ, start_response)
           │    │           │        └ <bound method BaseHandler.start_response of <django.core.servers.basehttp.ServerHandler object at 0x1051ddfa0>>
           │    │           └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
           │    └ <django.core.handlers.wsgi.WSGIHandler object at 0x1046c2720>
           └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x10545fb30>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
    response = self.get_response(request)
               │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function BaseHandler.get_response at 0x104b305e0>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1046c2720>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
               │    │                 └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │    └ <function convert_exception_to_response.<locals>.inner at 0x105513f60>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1046c2720>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513ec0>
               │           └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513d80>
               │           └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513ce0>
               │           └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513c40>
               │           └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513ba0>
               │           └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x105513b00>
               │           └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               │           │    └ <function BaseHandler._get_response at 0x104b885e0>
               │           └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
               └ None
> File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <bound method BaseHandler._get_response of <django.core.handlers.wsgi.WSGIHandler object at 0x1046c2720>>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               │                │         │                └ {}
               │                │         └ ()
               │                └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
               └ <function mongo_connection at 0x10548a8e0>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 53, in mongo_connection
    return render_toast_response(request)
           │                     └ <WSGIRequest: POST '/mongodb/mongo_connection/'>
           └ <function render_toast_response at 0x10548a840>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 17, in render_toast_response
    html = render_to_string("partials/toasts.html", {"messages": storage})
           │                                                     └ <FallbackStorage: request=<WSGIRequest: POST '/mongodb/mongo_connection/'>>
           └ <function render_to_string at 0x104238f40>

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               │            │                    └ None
               │            └ 'partials/toasts.html'
               └ <function get_template at 0x1040b74c0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
          │                    │                    └ [TemplateDoesNotExist('partials/toasts.html')]
          │                    └ 'partials/toasts.html'
          └ <class 'django.template.exceptions.TemplateDoesNotExist'>

django.template.exceptions.TemplateDoesNotExist: partials/toasts.html
2025-08-29 16:38:13.974 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:38:14.104 | INFO     | 
2025-08-29 16:38:14.104 | INFO     | --- Start APP ---
2025-08-29 16:38:14.138 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 16:38:25.259 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:38:25.365 | INFO     | 
2025-08-29 16:38:25.365 | INFO     | --- Start APP ---
2025-08-29 16:38:25.365 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:38:25.373 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:52:57.009 | INFO     | /Users/evgenij/Projects/WWS1/mongodb/views.py changed, reloading.
2025-08-29 16:52:57.295 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:54:38.323 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:54:38.459 | INFO     | 
2025-08-29 16:54:38.459 | INFO     | --- Start APP ---
2025-08-29 16:54:38.459 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:54:38.467 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:54:38.486 | ERROR    | Internal Server Error: /mongodb/mongo_connection/
Traceback (most recent call last):

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/smartif.py", line 180, in translate_token
    op = OPERATORS[token]
         │         └ 'request.headers.HX-Request'
         └ {'or': <class 'django.template.smartif.infix.<locals>.Operator'>, 'and': <class 'django.template.smartif.infix.<locals>.Opera...

KeyError: 'request.headers.HX-Request'


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1032, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x103fc5940>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6155317248)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x103fc5620>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6155317248)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6155317248)>
    │    │        │    └ (<socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50692)>, ('127.0.0.1', 50692))
    │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6155317248)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <django.core.servers.basehttp.WSGIServer object at 0x105c17980>>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6155317248)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 50692)
    │    │              └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50692)>
    │    └ <function BaseServer.finish_request at 0x105479260>
    └ <django.core.servers.basehttp.WSGIServer object at 0x105c17980>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <django.core.servers.basehttp.WSGIServer object at 0x105c17980>
    │    │                   │        └ ('127.0.0.1', 50692)
    │    │                   └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50692)>
    │    └ <class 'django.core.servers.basehttp.WSGIRequestHandler'>
    └ <django.core.servers.basehttp.WSGIServer object at 0x105c17980>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x1058d79c0>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x107e9fec0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 230, in handle
    self.handle_one_request()
    │    └ <function WSGIRequestHandler.handle_one_request at 0x1058d7a60>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x107e9fec0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 253, in handle_one_request
    handler.run(self.server.get_app())
    │       │   │    │      └ <function WSGIServer.get_app at 0x1058d6ac0>
    │       │   │    └ <django.core.servers.basehttp.WSGIServer object at 0x105c17980>
    │       │   └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x107e9fec0>
    │       └ <function BaseHandler.run at 0x1058ceac0>
    └ <django.core.servers.basehttp.ServerHandler object at 0x107c614f0>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/wsgiref/handlers.py", line 137, in run
    self.result = application(self.environ, self.start_response)
    │    │        │           │    │        │    └ <function BaseHandler.start_response at 0x1058cee80>
    │    │        │           │    │        └ <django.core.servers.basehttp.ServerHandler object at 0x107c614f0>
    │    │        │           │    └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
    │    │        │           └ <django.core.servers.basehttp.ServerHandler object at 0x107c614f0>
    │    │        └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x107b464b0>
    │    └ None
    └ <django.core.servers.basehttp.ServerHandler object at 0x107c614f0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/contrib/staticfiles/handlers.py", line 80, in __call__
    return self.application(environ, start_response)
           │    │           │        └ <bound method BaseHandler.start_response of <django.core.servers.basehttp.ServerHandler object at 0x107c614f0>>
           │    │           └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
           │    └ <django.core.handlers.wsgi.WSGIHandler object at 0x1052c5d00>
           └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x107b464b0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
    response = self.get_response(request)
               │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │    └ <function BaseHandler.get_response at 0x1058945e0>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1052c5d00>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
               │    │                 └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │    └ <function convert_exception_to_response.<locals>.inner at 0x107e3af20>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1052c5d00>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x107e3ae80>
               │           └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x107e3ade0>
               │           └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x107e3ad40>
               │           └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x107e3afc0>
               │           └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x107e3a2a0>
               │           └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x107e3aca0>
               │           └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function BaseHandler._get_response at 0x1058ec5e0>
               │           └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
               └ None
> File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <bound method BaseHandler._get_response of <django.core.handlers.wsgi.WSGIHandler object at 0x1052c5d00>>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               │                │         │                └ {}
               │                │         └ ()
               │                └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <function mongo_connection at 0x107c56840>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 86, in mongo_connection
    return render(request, 'mongodb/connection_form.html', {
           │      └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
           └ <function render at 0x10566dbc0>

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              │      │                │              │        │              └ None
              │      │                │              │        └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
              │      │                │              └ {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host;port)>, 'text': {'title': 'Serverkonfiguration', 'head...
              │      │                └ 'mongodb/connection_form.html'
              │      └ <function render_to_string at 0x104f9cf40>
              └ <module 'django.template.loader' from '/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader...
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           │        │      │        └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
           │        │      └ {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host;port)>, 'text': {'title': 'Serverkonfiguration', 'head...
           │        └ <function Template.render at 0x104f9e200>
           └ <django.template.backends.django.Template object at 0x1058e53a0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/backends/django.py", line 107, in render
    return self.template.render(context)
           │    │        │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           │    │        └ <function Template.render at 0x104f94d60>
           │    └ <Template template_string="{% extends 'base.htm...">
           └ <django.template.backends.django.Template object at 0x1058e53a0>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           │    │       └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           │    └ <function Template._render at 0x104f94cc0>
           └ <Template template_string="{% extends 'base.htm...">
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           │    │        │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           │    │        └ <function NodeList.render at 0x104f968e0>
           │    └ [<ExtendsNode: extends 'base.html'>]
           └ <Template template_string="{% extends 'base.htm...">
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
           │                                         │                    └ [<ExtendsNode: extends 'base.html'>]
           │                                         └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           └ <class 'django.utils.safestring.SafeString'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 977, in render_annotated
    return self.render(context)
           │    │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           │    └ <function ExtendsNode.render at 0x1055b0540>
           └ <ExtendsNode: extends 'base.html'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader_tags.py", line 134, in render
    compiled_parent = self.get_parent(context)
                      │    │          └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
                      │    └ <function ExtendsNode.get_parent at 0x1055b04a0>
                      └ <ExtendsNode: extends 'base.html'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader_tags.py", line 131, in get_parent
    return self.find_template(parent, context)
           │    │             │       └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           │    │             └ 'base.html'
           │    └ <function ExtendsNode.find_template at 0x1055b0400>
           └ <ExtendsNode: extends 'base.html'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader_tags.py", line 109, in find_template
    template, origin = context.template.engine.find_template(
                       │       └ None
                       └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               │      │            │          └ [<Origin name='/Users/evgenij/Projects/WWS1/templates/mongodb/connection_form.html'>]
               │      │            └ 'base.html'
               │      └ <function Loader.get_template at 0x107c57e20>
               └ <django.template.loaders.cached.Loader object at 0x1058e7b60>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loaders/cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
                                    │              └ [<Origin name='/Users/evgenij/Projects/WWS1/templates/mongodb/connection_form.html'>]
                                    └ 'base.html'
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loaders/base.py", line 28, in get_template
    return Template(
           └ <class 'django.template.base.Template'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
    │               │    └ <function Template.compile_nodelist at 0x104f94e00>
    │               └ <Template template_string="{% load static %}<!...">
    └ <Template template_string="{% load static %}<!...">
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               │      └ <function Parser.parse at 0x104f95800>
               └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
          │    │     └ <Block token: "if not request.heade...">
          │    └ <function Parser.error at 0x104f959e0>
          └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      │            │     └ <Block token: "if not request.heade...">
                      │            └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
                      └ <function do_if at 0x1055a9e40>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/defaulttags.py", line 961, in do_if
    condition = TemplateIfParser(parser, bits).parse()
                │                │       └ ['not', 'request.headers.HX-Request']
                │                └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
                └ <class 'django.template.defaulttags.TemplateIfParser'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/defaulttags.py", line 894, in __init__
    super().__init__(*args, **kwargs)
                      │       └ {}
                      └ (['not', 'request.headers.HX-Request'],)
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/smartif.py", line 171, in __init__
    mapped_tokens.append(self.translate_token(token))
    │             │      │    │               └ 'request.headers.HX-Request'
    │             │      │    └ <function IfParser.translate_token at 0x105583240>
    │             │      └ <django.template.defaulttags.TemplateIfParser object at 0x11100e240>
    │             └ <method 'append' of 'list' objects>
    └ [(not)]
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/smartif.py", line 182, in translate_token
    return self.create_var(token)
           │    │          └ 'request.headers.HX-Request'
           │    └ <function TemplateIfParser.create_var at 0x1055a9d00>
           └ <django.template.defaulttags.TemplateIfParser object at 0x11100e240>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/defaulttags.py", line 897, in create_var
    return TemplateLiteral(self.template_parser.compile_filter(value), value)
           │               │    │               │              │       └ 'request.headers.HX-Request'
           │               │    │               │              └ 'request.headers.HX-Request'
           │               │    │               └ <function Parser.compile_filter at 0x104f95e40>
           │               │    └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
           │               └ <django.template.defaulttags.TemplateIfParser object at 0x11100e240>
           └ <class 'django.template.defaulttags.TemplateLiteral'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 609, in compile_filter
    return FilterExpression(token, self)
           │                │      └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
           │                └ 'request.headers.HX-Request'
           └ <class 'django.template.base.FilterExpression'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 710, in __init__
    raise TemplateSyntaxError(
          └ <class 'django.template.exceptions.TemplateSyntaxError'>

django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '-Request' from 'request.headers.HX-Request'
2025-08-29 16:55:34.777 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:55:34.896 | INFO     | 
2025-08-29 16:55:34.896 | INFO     | --- Start APP ---
2025-08-29 16:55:34.896 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:55:34.902 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:55:34.918 | ERROR    | Internal Server Error: /mongodb/mongo_connection/
Traceback (most recent call last):

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/smartif.py", line 180, in translate_token
    op = OPERATORS[token]
         │         └ 'request.headers.HX-Request'
         └ {'or': <class 'django.template.smartif.infix.<locals>.Operator'>, 'and': <class 'django.template.smartif.infix.<locals>.Opera...

KeyError: 'request.headers.HX-Request'


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1032, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x104ce1940>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6136770560)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
    self.run()
    │    └ <function Thread.run at 0x104ce1620>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6136770560)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {}
    │    │        │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6136770560)>
    │    │        │    └ (<socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50698)>, ('127.0.0.1', 50698))
    │    │        └ <Thread(Thread-1 (process_request_thread), started daemon 6136770560)>
    │    └ <bound method ThreadingMixIn.process_request_thread of <django.core.servers.basehttp.WSGIServer object at 0x10713ec00>>
    └ <Thread(Thread-1 (process_request_thread), started daemon 6136770560)>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 692, in process_request_thread
    self.finish_request(request, client_address)
    │    │              │        └ ('127.0.0.1', 50698)
    │    │              └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50698)>
    │    └ <function BaseServer.finish_request at 0x10657d260>
    └ <django.core.servers.basehttp.WSGIServer object at 0x10713ec00>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 362, in finish_request
    self.RequestHandlerClass(request, client_address, self)
    │    │                   │        │               └ <django.core.servers.basehttp.WSGIServer object at 0x10713ec00>
    │    │                   │        └ ('127.0.0.1', 50698)
    │    │                   └ <socket.socket fd=7, family=2, type=1, proto=0, laddr=('127.0.0.1', 8000), raddr=('127.0.0.1', 50698)>
    │    └ <class 'django.core.servers.basehttp.WSGIRequestHandler'>
    └ <django.core.servers.basehttp.WSGIServer object at 0x10713ec00>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/socketserver.py", line 761, in __init__
    self.handle()
    │    └ <function WSGIRequestHandler.handle at 0x1069db9c0>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x10727fc20>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 230, in handle
    self.handle_one_request()
    │    └ <function WSGIRequestHandler.handle_one_request at 0x1069dba60>
    └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x10727fc20>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/servers/basehttp.py", line 253, in handle_one_request
    handler.run(self.server.get_app())
    │       │   │    │      └ <function WSGIServer.get_app at 0x1069daac0>
    │       │   │    └ <django.core.servers.basehttp.WSGIServer object at 0x10713ec00>
    │       │   └ <django.core.servers.basehttp.WSGIRequestHandler object at 0x10727fc20>
    │       └ <function BaseHandler.run at 0x1069d2ac0>
    └ <django.core.servers.basehttp.ServerHandler object at 0x10727fd40>
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/wsgiref/handlers.py", line 137, in run
    self.result = application(self.environ, self.start_response)
    │    │        │           │    │        │    └ <function BaseHandler.start_response at 0x1069d2e80>
    │    │        │           │    │        └ <django.core.servers.basehttp.ServerHandler object at 0x10727fd40>
    │    │        │           │    └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
    │    │        │           └ <django.core.servers.basehttp.ServerHandler object at 0x10727fd40>
    │    │        └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x10705b110>
    │    └ None
    └ <django.core.servers.basehttp.ServerHandler object at 0x10727fd40>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/contrib/staticfiles/handlers.py", line 80, in __call__
    return self.application(environ, start_response)
           │    │           │        └ <bound method BaseHandler.start_response of <django.core.servers.basehttp.ServerHandler object at 0x10727fd40>>
           │    │           └ {'HOMEBREW_PREFIX': '/opt/homebrew', 'COMMAND_MODE': 'unix2003', 'PYTHONUNBUFFERED': '1', 'INFOPATH': '/opt/homebrew/share/in...
           │    └ <django.core.handlers.wsgi.WSGIHandler object at 0x1063a4f80>
           └ <django.contrib.staticfiles.handlers.StaticFilesHandler object at 0x10705b110>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/wsgi.py", line 124, in __call__
    response = self.get_response(request)
               │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │    └ <function BaseHandler.get_response at 0x1069985e0>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1063a4f80>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = self._middleware_chain(request)
               │    │                 └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │    └ <function convert_exception_to_response.<locals>.inner at 0x10721ae80>
               └ <django.core.handlers.wsgi.WSGIHandler object at 0x1063a4f80>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x1071f7100>
               │           └ <SecurityMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x10721ade0>
               │           └ <SessionMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x1071f7ce0>
               │           └ <CommonMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x1069f3380>
               │           └ <CsrfViewMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x10721ad40>
               │           └ <AuthenticationMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function convert_exception_to_response.<locals>.inner at 0x10721a480>
               │           └ <MessageMiddleware get_response=convert_exception_to_response.<locals>.inner>
               └ None
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/utils/deprecation.py", line 120, in __call__
    response = response or self.get_response(request)
               │           │    │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               │           │    └ <function BaseHandler._get_response at 0x1069f05e0>
               │           └ <XFrameOptionsMiddleware get_response=BaseHandler._get_response>
               └ None
> File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               │            └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <bound method BaseHandler._get_response of <django.core.handlers.wsgi.WSGIHandler object at 0x1063a4f80>>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               │                │         │                └ {}
               │                │         └ ()
               │                └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
               └ <function mongo_connection at 0x10716a840>

  File "/Users/evgenij/Projects/WWS1/mongodb/views.py", line 86, in mongo_connection
    return render(request, 'mongodb/connection_form.html', {
           │      └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
           └ <function render at 0x106771bc0>

  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              │      │                │              │        │              └ None
              │      │                │              │        └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
              │      │                │              └ {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host;port)>, 'text': {'title': 'Serverkonfiguration', 'head...
              │      │                └ 'mongodb/connection_form.html'
              │      └ <function render_to_string at 0x1060a0f40>
              └ <module 'django.template.loader' from '/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader...
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           │        │      │        └ <WSGIRequest: GET '/mongodb/mongo_connection/'>
           │        │      └ {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host;port)>, 'text': {'title': 'Serverkonfiguration', 'head...
           │        └ <function Template.render at 0x1060a2200>
           └ <django.template.backends.django.Template object at 0x1072c5910>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/backends/django.py", line 107, in render
    return self.template.render(context)
           │    │        │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           │    │        └ <function Template.render at 0x106094d60>
           │    └ <Template template_string="{% extends 'base.htm...">
           └ <django.template.backends.django.Template object at 0x1072c5910>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           │    │       └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           │    └ <function Template._render at 0x106094cc0>
           └ <Template template_string="{% extends 'base.htm...">
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           │    │        │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           │    │        └ <function NodeList.render at 0x1060968e0>
           │    └ [<ExtendsNode: extends 'base.html'>]
           └ <Template template_string="{% extends 'base.htm...">
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
           │                                         │                    └ [<ExtendsNode: extends 'base.html'>]
           │                                         └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           └ <class 'django.utils.safestring.SafeString'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 977, in render_annotated
    return self.render(context)
           │    │      └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           │    └ <function ExtendsNode.render at 0x1066b4540>
           └ <ExtendsNode: extends 'base.html'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader_tags.py", line 134, in render
    compiled_parent = self.get_parent(context)
                      │    │          └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
                      │    └ <function ExtendsNode.get_parent at 0x1066b44a0>
                      └ <ExtendsNode: extends 'base.html'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader_tags.py", line 131, in get_parent
    return self.find_template(parent, context)
           │    │             │       └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
           │    │             └ 'base.html'
           │    └ <function ExtendsNode.find_template at 0x1066b4400>
           └ <ExtendsNode: extends 'base.html'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loader_tags.py", line 109, in find_template
    template, origin = context.template.engine.find_template(
                       │       └ None
                       └ [{'True': True, 'False': False, 'None': None}, {}, {}, {'form': <MongoConnectionForm bound=False, valid=Unknown, fields=(host...
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               │      │            │          └ [<Origin name='/Users/evgenij/Projects/WWS1/templates/mongodb/connection_form.html'>]
               │      │            └ 'base.html'
               │      └ <function Loader.get_template at 0x10716be20>
               └ <django.template.loaders.cached.Loader object at 0x10713ee40>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loaders/cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
                                    │              └ [<Origin name='/Users/evgenij/Projects/WWS1/templates/mongodb/connection_form.html'>]
                                    └ 'base.html'
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/loaders/base.py", line 28, in get_template
    return Template(
           └ <class 'django.template.base.Template'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
    │               │    └ <function Template.compile_nodelist at 0x106094e00>
    │               └ <Template template_string="{% load static %}<!...">
    └ <Template template_string="{% load static %}<!...">
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               │      └ <function Parser.parse at 0x106095800>
               └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 518, in parse
    raise self.error(token, e)
          │    │     └ <Block token: "if not request.heade...">
          │    └ <function Parser.error at 0x1060959e0>
          └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      │            │     └ <Block token: "if not request.heade...">
                      │            └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
                      └ <function do_if at 0x1066ade40>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/defaulttags.py", line 961, in do_if
    condition = TemplateIfParser(parser, bits).parse()
                │                │       └ ['not', 'request.headers.HX-Request']
                │                └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
                └ <class 'django.template.defaulttags.TemplateIfParser'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/defaulttags.py", line 894, in __init__
    super().__init__(*args, **kwargs)
                      │       └ {}
                      └ (['not', 'request.headers.HX-Request'],)
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/smartif.py", line 171, in __init__
    mapped_tokens.append(self.translate_token(token))
    │             │      │    │               └ 'request.headers.HX-Request'
    │             │      │    └ <function IfParser.translate_token at 0x106687240>
    │             │      └ <django.template.defaulttags.TemplateIfParser object at 0x107332390>
    │             └ <method 'append' of 'list' objects>
    └ [(not)]
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/smartif.py", line 182, in translate_token
    return self.create_var(token)
           │    │          └ 'request.headers.HX-Request'
           │    └ <function TemplateIfParser.create_var at 0x1066add00>
           └ <django.template.defaulttags.TemplateIfParser object at 0x107332390>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/defaulttags.py", line 897, in create_var
    return TemplateLiteral(self.template_parser.compile_filter(value), value)
           │               │    │               │              │       └ 'request.headers.HX-Request'
           │               │    │               │              └ 'request.headers.HX-Request'
           │               │    │               └ <function Parser.compile_filter at 0x106095e40>
           │               │    └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
           │               └ <django.template.defaulttags.TemplateIfParser object at 0x107332390>
           └ <class 'django.template.defaulttags.TemplateLiteral'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 609, in compile_filter
    return FilterExpression(token, self)
           │                │      └ <Parser tokens=[<Text token: "</body></html>...">, <Block token: "endblock...">, <Block token: "block extra_js...">, <Text to...
           │                └ 'request.headers.HX-Request'
           └ <class 'django.template.base.FilterExpression'>
  File "/Users/evgenij/Projects/WWS1/.venv/lib/python3.12/site-packages/django/template/base.py", line 710, in __init__
    raise TemplateSyntaxError(
          └ <class 'django.template.exceptions.TemplateSyntaxError'>

django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '-Request' from 'request.headers.HX-Request'
2025-08-29 16:58:35.218 | INFO     | Watching for file changes with StatReloader
2025-08-29 16:58:35.347 | INFO     | 
2025-08-29 16:58:35.347 | INFO     | --- Start APP ---
2025-08-29 16:58:35.347 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:58:35.353 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:58:47.478 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1c022a9b9f6e4fa76b713, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 16:59:06.911 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 16:59:09.097 | SUCCESS  | Ping des MongoDB-Servers war erfolgreich
2025-08-29 16:59:09.103 | INFO     | Konfigurationsdatei für die Datenbankverbindung aktualisiert
2025-08-29 16:59:09.103 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 17:00:52.315 | INFO     | Watching for file changes with StatReloader
2025-08-29 17:00:52.444 | INFO     | 
2025-08-29 17:00:52.444 | INFO     | --- Start APP ---
2025-08-29 17:00:52.479 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 17:01:05.666 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27018: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1c0acf823e71b0f410eac, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27018: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 17:02:29.478 | INFO     | Watching for file changes with StatReloader
2025-08-29 17:02:29.577 | INFO     | 
2025-08-29 17:02:29.578 | INFO     | --- Start APP ---
2025-08-29 17:02:29.613 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 17:02:42.172 | INFO     | Watching for file changes with StatReloader
2025-08-29 17:02:42.276 | INFO     | 
2025-08-29 17:02:42.277 | INFO     | --- Start APP ---
2025-08-29 17:02:42.277 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 17:02:42.284 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 17:02:50.857 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1c115a6d907b85d42caaf, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 17:05:42.052 | INFO     | Watching for file changes with StatReloader
2025-08-29 17:05:42.174 | INFO     | 
2025-08-29 17:05:42.174 | INFO     | --- Start APP ---
2025-08-29 17:05:42.174 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 17:05:42.180 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 17:05:54.943 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1c1cdb570706a4da03442, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 18:08:50.759 | INFO     | Watching for file changes with StatReloader
2025-08-29 18:08:50.912 | INFO     | 
2025-08-29 18:08:50.913 | INFO     | --- Start APP ---
2025-08-29 18:08:50.913 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 18:08:50.917 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 18:09:03.778 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1d09aa5bc3ee0b3afb74e, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 18:30:27.355 | INFO     | /Users/evgenij/Projects/WWS1/WWS1/settings.py changed, reloading.
2025-08-29 18:31:45.585 | INFO     | Watching for file changes with StatReloader
2025-08-29 18:31:45.734 | INFO     | 
2025-08-29 18:31:45.735 | INFO     | --- Start APP ---
2025-08-29 18:31:45.735 | ERROR    | Ошибки в конфигурации:
2025-08-29 18:31:45.735 | ERROR    |   - Отсутствует обязательное поле: host
2025-08-29 18:31:45.735 | ERROR    |   - Отсутствует обязательное поле: port
2025-08-29 18:31:45.735 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 18:31:45.739 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 18:31:57.871 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1d5f828afdd7c2c612587, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27016) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27016: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 18:40:05.562 | INFO     | /Users/evgenij/Projects/WWS1/mongodb/views.py changed, reloading.
2025-08-29 18:40:05.751 | INFO     | Watching for file changes with StatReloader
2025-08-29 18:40:52.725 | INFO     | Watching for file changes with StatReloader
2025-08-29 18:40:52.833 | INFO     | 
2025-08-29 18:40:52.833 | INFO     | --- Start APP ---
2025-08-29 18:40:52.833 | ERROR    | Ошибки в конфигурации:
2025-08-29 18:40:52.833 | ERROR    |   - Отсутствует обязательное поле: host
2025-08-29 18:40:52.833 | ERROR    |   - Отсутствует обязательное поле: port
2025-08-29 18:40:52.833 | ERROR    | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 18:40:52.841 | WARNING  | Parameter zum Pingen des Datenbankservers fehlen oder sind falsch.
2025-08-29 18:41:08.350 | ERROR    | Ping des MongoDB-Server fehlgeschlagen: ef-soft.local:27018: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms), Timeout: 5.0s, Topology Description: <TopologyDescription id: 68b1d81faf46d0bad283ab5f, topology_type: Unknown, servers: [<ServerDescription ('ef-soft.local', 27018) server_type: Unknown, rtt: None, error=AutoReconnect('ef-soft.local:27018: [Errno 61] Connection refused (configured timeouts: socketTimeoutMS: 20000.0ms, connectTimeoutMS: 20000.0ms)')>]>
2025-08-29 18:41:29.106 | SUCCESS  | Ping des MongoDB-Servers war erfolgreich
2025-08-29 18:41:29.112 | INFO     | Konfigurationsdatei für die Datenbankverbindung aktualisiert
2025-08-29 18:41:29.112 | SUCCESS  | ef-soft.local:27017 — Ping des MongoDB-Servers war erfolgreich
2025-08-29 18:54:46.585 | INFO     | /Users/evgenij/Projects/WWS1/manage.py changed, reloading.
